app/models/user.rb

```diff
@@ -0,0 +1,7 @@
+class User < ActiveRecord::Base
+  has_many :lists
+
+  validates :name, length: { minimum: 1, maximum: 20}, presence: true
```

Comment:
These length ones I would validate in rails, but presence and uniqueness I would validate at the database level.